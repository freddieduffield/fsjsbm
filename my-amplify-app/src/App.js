import React, {Component} from 'react';

import './App.css';

import {withAuthenticator} from 'aws-amplify-react'
// import { API } from 'aws-amplify'
import { storage } from 'aws-amplify'


// const ListTodos = ` 
//   query { 
//     listTodos {
//       items { 
//         id name description completed 
//       }
//     }
//   }`

  class App extends Component {
    state = {fileUrl: '', file:'', filename: ''}

    handleChange = e => {
      const file = e.target.files[0]
      this.setState({
        fileUrl: URL.createObjectURL(file),
        file,
        filename: file.name
      })
    }

    saveFile = () => {
      Storage.put(this.state.filename, this.state.file)
        .then(() => {

          console.log('successfully saved file!');
          this.setState({fileUrl: '', file: '', filename: ''})
        }
          )
        .catch(err => {
          console.log('error updating file', err);
        })
    }

    // async componentDidMount() {
    //   const data = await API.get('peopleapi', '/people')
    //   this.setState({people: data.people})
    // }
    render() {
      return (
        <div>
          {/* {
            this.state.people.map((person, i) => (
              <div>
                <h3>{person.name}</h3>
                <p>{person.hair_color}</p>
              </div>
            ))
          } */}

          <input type="file" onChange={this.handleChange} /> 
          <img src={this.state.fileUrl} />
          <button onClick={this.saveFile}>Save File</button>
        </div>
      )
    }
  }


// class App extends Component {
 
//   state = {username: '', password: '', phone_number: '', authenticationCode: '', step : 0}

//   onChange = e => {
//     this.setState({[e.target.name]: e.target.value})
//   }

//   signUp = async () => {
//     const {username, password, email, phone_number} = this.state
//     try {
//       await Auth.signUp({username, password, attributes: {email, phone_number  }})
//       console.log('successfully signed up')
//       this.setState({step: 1}) 
//     } catch (err) {console.log('error signing up: ', err)}
//   }

//   confirmSignup = async () => {
//     const {username, authenticationCode} = this.state
//     try {
//       await Auth.confirmSignUp(username, authenticationCode)
//       console.log('user successfully signed up')
//     } catch (err) {console.log('error confirming sign up : ', err)}
//   }

//   render() {
//     return (
//     <div className="App">
//       {
//         this.state.step === 0 && (
//           <div>
//             <input placeholder="username" onChange={this.onChange} name="username" styles={styles.input} />
//             <input placeholder="password" onChange={this.onChange} name="password" type="password" styles={styles.input} />
//             <input placeholder="email" onChange={this.onChange} name="email" styles={styles.input} />
//             <input placeholder="phone number" onChange={this.onChange} name="phone_number" styles={styles.input} />
//             <button onClick={this.signUp}>Sign Up</button>
//           </div>
//         )
//       }
//       {
//         this.state.step === 1 && (
//           <div>
//             <input placeholder="username" onChange={this.onChange} name="username" styles={styles.input} />
//             <input placeholder="authentication code" onChange={this.onChange} name="authenticationCode" type="password" styles={styles.input} />
//             <button onClick={this.confirmSignUp}>Confirm Sign Up</button>
//           </div>
//         )
//       }
//     </div>);

//   }
// }

const styles = {
  input: {
    height: 35, margin: 100
  }
}


export default withAuthenticator(App);
