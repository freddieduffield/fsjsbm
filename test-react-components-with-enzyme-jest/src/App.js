import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>;
const Title = ({ text }) => <div>{text}</div>;

class App extends Component {
  state = {
    on: false,
    input: ''
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Title text="Some title" />
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({ on: true })}>Click</button>
        <h2>{this.state.input}</h2>
        <input
          onChange={e => this.setState({ input: e.currentTarget.value })}
          type="text"
        />
      </div>
    );
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
}

export default App;
