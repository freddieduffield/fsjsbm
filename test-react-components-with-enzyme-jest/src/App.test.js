import React from 'react';
import App, { Link } from './App';

import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

/* first test */
// describe('<App />', () => {
//   it('should render App', () => {
//     const wrapper = shallow(<App />, {
//       context: {},
//       disableLifecycleMethods: bool
//     });
//     console.log(wrapper.debug());
//   });
// });

/* 
 Find nodes from a shallow rendered component
*/

// describe('<App />', () => {
//   const wrapper = shallow(<App />);

//   it('should have a list with three children', () => {
//     expect(wrapper.find('ul').children().length).toBe(3);
//   });

//   it('should have a list with a class of fred', () => {
//     expect(wrapper.find('ul').hasClass('fred')).toBe(true);
//   });

//   it('should have 1 p tag', () => {
//     expect(wrapper.find('p').length).toBe(1);
//   });

//   it('should have 1 p tag', () => {
//     expect(wrapper.find('p').exists()).toBe(true);
//   });

//   it('should have 1 p tag', () => {
//     expect(wrapper.find('.App-intro').exists()).toBe(true);
//   });

//   it('should have h1 with the text Welcome to React ', () => {
//     expect(wrapper.find('h1').text()).toBe('Welcome to React');
//   });
// });

// children()
// length
// find('SELECTOR')
// exists()
// hasClass('className')
// text()

/*
 Selectors
 */

//  CSS selectors
// props
// function
// objects

// describe('<App />', () => {
//   const wrapper = shallow(<App />);

//   it('should contain h1 with correct text', () => {
//     expect(wrapper.find('[text="Some title"]').text()).toBe('Some title');
//   });
// });

/* 
snapshots
*/

// describe('<App />', () => {

//   it('matches the snapshot', () => {
//     const tree = shallow(<App />)
//     expect(toJson(tree)).toMatchSnapshot()
//   });

// });

/*
Test React Component Props with Enzyme and Jest
*/

describe('<Link />', () => {
  it('link component accepts address prop', () => {
    const wrapper = shallow(<Link address="www.google.com" />);
    expect(wrapper.instance().props.address).toBe('www.google.com');
  });

  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address="www.google.com" />);
    expect(wrapper.props().href).toBe('www.google.com');
  });

  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull;
  });
});

/*
Fully render React Components with Enzyme
*/

describe('<App /> mount rendering', () => {
  it('matches the snapshot', () => {
    const tree = mount(<App />);
    expect(toJson(tree)).toMatchSnapshot();

    tree.unmount();
  });

  it('should have h1 with the text Welcome to React ', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
    wrapper.unmount();
  });
});

describe('<App /> shallow rendering', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('updates className with new State', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.blue').length).toBe(1);
    expect(wrapper.find('.red').length).toBe(0);
    wrapper.setState({ mainColor: 'red' });
    expect(wrapper.find('.red').length).toBe(1);
    expect(wrapper.find('.blue').length).toBe(0);
  });

  it('should have h1 with the text Welcome to React ', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('h1').text()).toBe('Welcome to React');
  });

  it('on button click change p text', () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('No!');
    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('Yes!');
  });

  it('on input change, title changes text', () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find('input');

    expect(wrapper.find('h2').text()).toBe('');
    input.simulate('change', { currentTarget: { value: 'Tyler' } });
    expect(wrapper.find('h2').text()).toBe('Tyler');
  });

  it('calls componentDidMount, updates P tag text', () => {
    jest.spyOn(App.prototype, 'componentDidMount');

    const wrapper = shallow(<App />);

    expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(wrapper.find('.lifeCycle').text()).toBe('componentDidMount');
  });

  it('setProps calls componentWillRecieveProps', () => {
    jest.spyOn(App.prototype, 'componentWillReceiveProps');
    const wrapper = shallow(<App />);
    wrapper.setProps({ hide: true });
    expect(App.prototype.componentWillReceiveProps.mock.calls.length).toBe(1);
    expect(wrapper.find('.lifeCycle').text()).toBe('componentWillReceiveProps');
  });
});
