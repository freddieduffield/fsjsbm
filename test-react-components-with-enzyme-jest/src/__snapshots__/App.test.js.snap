// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <Title
      text="Some title"
    >
      <div>
        Some title
      </div>
    </Title>
    <p
      className="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <Title
    text="Some title"
  />
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
</div>
`;
